hazelcast/src/main/java/com/hazelcast/query/impl/OrderedIndexStore.java

```diff
@@ -544,15 +545,7 @@ private static class DataComparator implements Comparator<Data> {
         public int compare(Data o1, Data o2) {
             byte[] thisBytes = o1.toByteArray();
             byte[] thatBytes = o2.toByteArray();
-            int minLen = Math.min(thisBytes.length, thatBytes.length);
-            for (int i = 0; i < minLen; i++) {
-                int diff = thisBytes[i] - thatBytes[i];
-                if (diff == 0) {
-                    continue;
-                }
-                return diff;
-            }
-            return thisBytes.length - thatBytes.length;
+            return Arrays.compare(thisBytes, thatBytes);
```

Comment:
`Arrays.compare` has null checks, which are redundant here. I wonder if that matters.